tests/test-systray.lua

```diff
@@ -81,6 +81,58 @@ table.insert(steps, function()
     return true
 end)
 
+table.insert(steps, function()
+    if draw_w ~= 100 then return end
+    return true
+end)
+
+table.insert(steps, function()
+    st.base_size = 20
+    pid1 = spawn("env USE_TRAY_VISUAL_HINT=1 ./test-systray")
+    return true
+end)
+
+local lgi_core = require("lgi.core")
+local lgi_ffi = require("lgi.ffi")
+local lgi_ti = lgi_ffi.types
+local lgi_record = require("lgi.record")
+local lgi_component = require("lgi.component")
+local cairo = require("lgi").cairo
+local wrapped_uchar = lgi_component.create(nil, lgi_record.struct_mt, "wrapped_uchar")
+lgi_ffi.load_fields(wrapped_uchar, { { 'v', lgi_ti.uchar } })
+
+table.insert(steps, function()
+    if draw_w ~= 80 then return end
+    local systray_surface_raw = awesome.systray_surface(20, 20)
+    if systray_surface_raw then
+        local src = cairo.Surface(systray_surface_raw, true)
+        local s = cairo.ImageSurface("ARGB32", 20, 20)
+        local cr = cairo.Context(s)
+        cr:set_source_surface(src, 0, 0)
+        cr:paint()
+        -- Read the first pixel (a,r,g,b) as 4 8-bit integers.
+        local data = s:get_data()
+        local array = lgi_core.record.new(wrapped_uchar, data, false)
+        local argb = {}
+        for i = 0, 3 do
+            argb[i + 1] = lgi_core.record.fromarray(array, i).v
+        end
+        -- Check with the pixel from test-systray.c
+        return
```

Comment:
Dunno if I got this right, but I think all this LGI magic is not necessary:
```
local data = s:get_data()
local function get_pixel(index) return string.byte(data, index) end
return get_pixel(1) == 0 and get_pixel(2) == 153 and ...
```
(Dunno if I got the indexes right, but it's the idea that counts here)

Reply:
Unfortunately surface.get_data returns a userdata that cannot be accessed by string.byte. That's why using the magic to "cast" it into an accessible LGI type.

Reply:
Hm, sorry. I thought that worked somehow. I also thought there was already some code somewhere working with `.data`, but I can't find it. Sorry!